client/client-lib/src/mint/mod.rs

```diff
@@ -555,14 +555,14 @@ mod tests {
         let mut dbtx = client_db.begin_transaction();
         client.receive_coins(amt, &mut dbtx, rng, |output| {
             // Agree on output
-            let mut fed = block_on(fed.lock());
-            block_on(fed.consensus_round(&[], &[(out_point, output)]));
+            let mut fed = async_std::task::block_on(fed.lock());
```

Comment:
`async_std`? I thought we use `tokio` for everything.

Reply:
I ran into issues trying to use futures::executor::block_on here (and other spots where I used async_std). 

Here's an example
thread 'ecash_in_wallet_can_sent_through_a_tx' panicked at 'cannot execute `LocalPool` executor from within another executor: EnterError', /home/jumoell/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-executor-0.3.25/src/local_pool.rs:81:26

I believe in this example the closure passed to receive_coins has a block_on and the function inside the block_on also calls block_on (to commit to the database). I got around this by using async_std but it doesn't feel like the right solution, open to suggestions.

Reply:
I'd have to dig into the code, but I just had to deal with `spawn_local` for spawning on `!Send` future: https://github.com/fedimint/fedimint/pull/849/files#diff-28cee3cca4f2764a431519d61c916039c91f61734a26d387698d34f627a15200R91

Importantly I needed to create a locatset, enter it, and then await on the same thread till completion.

Reply:
Going to try changing receive_coins to take an async closure as a parameter

Reply:
Would be nice if we could avoid pulling in `async-std`.

Reply:
Removed async-std dependency